upload-chunked-file.js

```diff
@@ -0,0 +1,43 @@
+function chunkFileUpload(file, serverUrl, chunkSize) {
+    const totalChunks = Math.ceil(file.size / chunkSize);
+    let chunkIndex = 0;
+
+    (function uploadNextChunk() {
```

Comment:
Consider using `const` and `let` instead of `var` for variable declarations. `const` is suitable for variables that do not change, and `let` is block-scoped and preferable for iterable variables like `chunkIndex`.